datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/role/BatchAssignRoleToActorsResolver.java

```diff
@@ -0,0 +1,89 @@
+import com.linkedin.datahub.graphql.exception.AuthorizationException;
+import com.linkedin.datahub.graphql.generated.BatchAssignRoleToActorsInput;
+import com.linkedin.entity.client.EntityClient;
+import com.linkedin.events.metadata.ChangeType;
+import com.linkedin.identity.RoleMembership;
+import com.linkedin.metadata.utils.GenericRecordUtils;
+import com.linkedin.mxe.MetadataChangeProposal;
+import com.linkedin.r2.RemoteInvocationException;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import static com.linkedin.datahub.graphql.authorization.AuthorizationUtils.*;
+import static com.linkedin.datahub.graphql.resolvers.ResolverUtils.*;
+import static com.linkedin.metadata.Constants.*;
+
+
+@Slf4j
+@RequiredArgsConstructor
+public class BatchAssignRoleToActorsResolver implements DataFetcher<CompletableFuture<Boolean>> {
+  private final EntityClient _entityClient;
+
+  @Override
+  public CompletableFuture<Boolean> get(DataFetchingEnvironment environment) throws Exception {
+    final QueryContext context = environment.getContext();
+    if (!canManagePolicies(context)) {
+      throw new AuthorizationException(
+          "Unauthorized to assign roles. Please contact your DataHub administrator if this needs corrective action.");
+    }
+
+    final BatchAssignRoleToActorsInput input =
+        bindArgument(environment.getArgument("input"), BatchAssignRoleToActorsInput.class);
+    final String roleUrnStr = input.getRoleUrn();
+    final List<String> actors = input.getActors();
+    final Authentication authentication = context.getAuthentication();
+
+    return CompletableFuture.supplyAsync(() -> {
+      try {
+        Urn roleUrn = Urn.createFromString(roleUrnStr);
+        if (!_entityClient.exists(roleUrn, authentication)) {
+          throw new RuntimeException(String.format("Role %s does not exist", roleUrnStr));
+        }
+
+        actors.forEach(actor -> {
+          try {
+            assignRoleToActor(actor, roleUrn, authentication);
+          } catch (Exception e) {
+            log.error(String.format("Failed to assign role to actor %s", actor), e);
+          }
+        });
+        return true;
+      } catch (Exception e) {
+        log.error("Failed to perform update against input {}, {}", input, e.getMessage());
+        throw new RuntimeException(String.format("Failed to perform update against input %s", input), e);
+      }
+    });
+  }
+
+  private void assignRoleToActor(String actor, Urn roleUrn, Authentication authentication)
+      throws URISyntaxException, RemoteInvocationException {
+    Urn actorUrn = Urn.createFromString(actor);
+    if (!_entityClient.exists(actorUrn, authentication)) {
+      log.error(String.format("Actor %s does not exist", actor));
```

Comment:
So we just skip this actor? Let's sau that..

Reply:
Let's do a log.warn actually. Also add a bit more context: "Failed to assign actor to role <role urn>. Actor does not exist."